### PR TITLE
Feat: change the default workflow step with deploy type

### PR DESCRIPTION
**Pull request by barnettZQG**
Signed-off-by: barnettZQG <barnett.zqg@gmail.com>

### Description of your changes

There are some changes:

* For automatically generated workflow steps in VelaUX, replace deploy2env with deploy, allowing users to modify them.
* For applications with the common components and cloud service components, The workflow first delivers the cloud service using the `deploy-cloud-resource` step.
* For automatically generated policies, will save to DB and allows users to modify them. 
* Ignore the components with terraform workload in the deploy step.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

This is an example application spec:

```yaml
spec:
  components:
  - name: hhh
    properties:
      chart: kubeapps
      repoType: helm
      secretRef: ""
      url: https://charts.bitnami.com/bitnami
      values:
        apprepository:
          containerSecurityContext:
            runAsNonRoot: true
      version: 8.0.16
    type: helm
  - name: hhh-rds
    properties:
      exposeType: ClusterIP
      password: dasdasd
    type: alibaba-rds
  policies:
  - name: bj
    properties:
      clusters:
      - local
      namespace: test
    type: topology
  - name: cloud
    properties:
      clusters:
      - bj
      namespace: cloud
    type: topology
  - name: bj2
    properties:
      clusters:
      - bj
      namespace: bj2
    type: topology
  - name: default
    properties:
      clusters:
      - local
      namespace: default
    type: topology
  - name: env-bindings-default
    properties:
      envs:
      - name: bj
        patch:
          components:
          - name: hhh-rds
            properties:
              providerRef:
                name: default
              writeConnectionSecretToRef:
                name: hhh-rds-default
                namespace: default
            type: alibaba-rds
        placement:
          clusterSelector:
            name: local
          namespaceSelector:
            name: test
      - name: bj2
        patch:
          components:
          - name: hhh-rds
            properties:
              providerRef:
                name: ""
              writeConnectionSecretToRef:
                name: hhh-rds-default
                namespace: default
            type: alibaba-rds
        placement:
          clusterSelector:
            name: bj
          namespaceSelector:
            name: bj2
      - name: default
        patch:
          components:
          - name: hhh-rds
            properties:
              providerRef:
                name: default
              writeConnectionSecretToRef:
                name: hhh-rds-default
                namespace: default
            type: alibaba-rds
        placement:
          clusterSelector:
            name: local
          namespaceSelector:
            name: default
    type: env-binding
  workflow:
    steps:
    - name: default-cloud-resource
      properties:
        env: default
        policy: env-bindings-default
      type: deploy-cloud-resource
    - name: bj2-cloud-resource
      properties:
        env: bj2
        policy: env-bindings-default
      type: deploy-cloud-resource
    - name: bj-cloud-resource
      properties:
        env: bj
        policy: env-bindings-default
      type: deploy-cloud-resource
    - name: bj
      properties:
        policies:
        - bj
      type: deploy
    - name: default
      properties:
        policies:
        - default
      type: deploy
    - name: bj2
      properties:
        policies:
        - bj2
      type: deploy
```

### CODE REVIEW

**Review by Somefive — APPROVED** [review]
LGTM, great job!

File: test/e2e-apiserver-test/application_test.go [hunk 1]
```diff
@@ -206,13 +206,13 @@ var _ = Describe("Test application rest api", func() {
 		var req2 = apisv1.CreatePolicyRequest{
 			Name:        "test2",
 			Description: "this is a test2 policy",
-			Type:        "wqsdasd",
-			Properties:  `{"image": "busybox","cmd":["sleep", "1000"],"lives": "3","enemies": "alien"}`,
+			Type:        "overide",
```
Comment (wonderflow) [inline test/e2e-apiserver-test/application_test.go:209]:
```suggestion
			Type:        "override",
```

File: pkg/workflow/providers/multicluster/deploy.go [hunk 2]
```diff
@@ -96,13 +101,22 @@ func selectPolicies(policies []v1beta1.AppPolicy, policyNames []string) ([]v1bet
...
-func loadComponents(ctx context.Context, cli client.Client, af *appfile.Appfile, components []common.ApplicationComponent) ([]common.ApplicationComponent, error) {
+func loadComponents(ctx context.Context, renderer oamProvider.WorkloadRenderer, cli client.Client, af *appfile.Appfile, components []common.ApplicationComponent, ignoreTerraformComponent bool) ([]common.ApplicationComponent, error) {
 	var loadedComponents []common.ApplicationComponent
 	for _, comp := range components {
 		loadedComp, err := af.LoadDynamicComponent(ctx, cli, comp.DeepCopy())
 		if err != nil {
 			return nil, err
 		}
+		if ignoreTerraformComponent {
+			wl, err := renderer(comp)
```
Comment (wonderflow) [inline pkg/workflow/providers/multicluster/deploy.go:112]: can we use comp.Type to judge here? so we don't need to render, this is a complex calculate which affect the performance

Comment (barnettZQG) [reply pkg/workflow/providers/multicluster/deploy.go:112]: The component type is diverse and irregular

**Review by wonderflow — APPROVED** [review]

**Review by yangsoon — APPROVED** [review]

**Review by FogDong — APPROVED** [review]

**Pull request merged**